internal/bgp/bgp.go

```diff
@@ -168,7 +168,15 @@ func (s *Session) connect() error {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	deadline, _ := ctx.Deadline()
-	conn, err := dialMD5(ctx, s.addr, s.password)
+
+	laddr := "[::]:0"
+	if (s.routerID.To4() != nil) {
+		laddr = fmt.Sprintf("%s:0", s.routerID.String())
+	} else if (s.routerID.To16() != nil) {
```

Comment:
Router IDs cannot be IPv6 addresses in BGP, so I think you can remove this branch. `s.routerID != nil` should imply `To4() != nil`